src/core/sagas/products.ts

```diff
@@ -9,26 +11,71 @@ import * as utils from '../utils';
 import { Tasks as productDetailsTasks } from './product-details';
 
 export namespace Tasks {
-  export function* fetchProducts(flux: FluxCapacitor, action: Actions.FetchProducts) {
+  export function* fetchProductsAndNavigations(flux: FluxCapacitor, action: Actions.FetchProducts) {
     try {
-      const request = yield effects.select(Requests.search, flux.config);
-      const res = yield effects.call([flux.clients.bridge, flux.clients.bridge.search], request);
-
-      if (res.redirect) {
-        yield effects.put(flux.actions.receiveRedirect(res.redirect));
+      const [products, navigations]: [Results, Store.Recommendations.Navigation[]] = yield effects.all([
```

Comment:
do we need to explicitly set the type here?

Reply:
They're `any` if we don't set the type.